sdk/tests/sdk/whirlpools/quote/increase-liquidity-quote.test.ts

```diff
@@ -0,0 +1,135 @@
+  const tokenMintA = new PublicKey("orcaEKTdK7LKz57vaAYr9QeNsVEPfiu6QeMU1kektZE");
+  const tokenMintB = new PublicKey("EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v");
+
+  it("sqrtPrice on lower bound, tokenB input", async () => {
+    const quote = increaseLiquidityQuoteByInputTokenWithParams({
+      inputTokenAmount: new BN(1000),
+      inputTokenMint: tokenMintB,
+      sqrtPrice: PriceMath.tickIndexToSqrtPriceX64(0),
+      tokenMintA,
+      tokenMintB,
+      tickLowerIndex: 0,
+      tickUpperIndex: 64,
+      tickCurrentIndex: 0,
+      slippageTolerance: Percentage.fromFraction(0, 100),
+    });
+
+    assert.ok(quote.liquidityAmount.isZero());
+    assert.ok(quote.tokenEstA.isZero());
+    assert.ok(quote.tokenEstB.isZero());
+    assert.ok(quote.tokenMaxA.isZero());
+    assert.ok(quote.tokenMaxB.isZero());
+  });
+
+  it("sqrtPrice on lower bound, tokenA input", async () => {
+    const quote = increaseLiquidityQuoteByInputTokenWithParams({
+      inputTokenAmount: new BN(1000),
+      inputTokenMint: tokenMintA,
+      sqrtPrice: PriceMath.tickIndexToSqrtPriceX64(0),
+      tokenMintA,
+      tokenMintB,
+      tickLowerIndex: 0,
+      tickUpperIndex: 64,
+      tickCurrentIndex: 0,
+      slippageTolerance: Percentage.fromFraction(0, 100),
+    });
+
+    assert.ok(quote.liquidityAmount.gtn(0));
+    assert.ok(quote.tokenEstA.gtn(0));
+    assert.ok(quote.tokenEstB.isZero());
+    assert.ok(quote.tokenMaxA.gtn(0));
+    assert.ok(quote.tokenMaxB.isZero());
+  });
+
+  it("tickCurrentIndex on lower bound but sqrtPrice not on lower bound, tokenA input", async () => {
+    assert.ok(PriceMath.tickIndexToSqrtPriceX64(1).subn(1).gt(PriceMath.tickIndexToSqrtPriceX64(0)));
+
+    const quote = increaseLiquidityQuoteByInputTokenWithParams({
+      inputTokenAmount: new BN(1000),
+      inputTokenMint: tokenMintA,
+      sqrtPrice: PriceMath.tickIndexToSqrtPriceX64(1).subn(1),
+      tokenMintA,
+      tokenMintB,
+      tickLowerIndex: 0,
+      tickUpperIndex: 64,
+      tickCurrentIndex: 0,
+      slippageTolerance: Percentage.fromFraction(0, 100),
+    });
+
+    assert.ok(quote.liquidityAmount.gtn(0));
+    assert.ok(quote.tokenEstA.gtn(0));
+    assert.ok(quote.tokenEstB.gtn(0));
+    assert.ok(quote.tokenMaxA.gtn(0));
+    assert.ok(quote.tokenMaxB.gtn(0));
+  });
+
+  it("tickCurrentIndex on lower bound but sqrtPrice not on lower bound, tokenB input", async () => {
+    assert.ok(PriceMath.tickIndexToSqrtPriceX64(1).subn(1).gt(PriceMath.tickIndexToSqrtPriceX64(0)));
+
+    const quote = increaseLiquidityQuoteByInputTokenWithParams({
+      inputTokenAmount: new BN(1000),
+      inputTokenMint: tokenMintB,
+      sqrtPrice: PriceMath.tickIndexToSqrtPriceX64(1).subn(1),
+      tokenMintA,
+      tokenMintB,
+      tickLowerIndex: 0,
+      tickUpperIndex: 64,
+      tickCurrentIndex: 0,
+      slippageTolerance: Percentage.fromFraction(0, 100),
+    });
+
+    assert.ok(quote.liquidityAmount.gtn(0));
```

Comment:
Guess this is the edge case that the old method failed on. This should be a non-zero-liquidity quote but because the other method was using tickIndexes for current price it would get a sqrtDiff of 0. Here you'd get a non-zero sqrtDiff which means we can get a valid quote